### ISSUE TITLE
Failed to url decode response, err: invalid URL escape "% o"

### ISSUE THREAD

**Issue by viktorbenei** [bitrise-io/steps-random-quote#7]
```
WARN[03:48:16] step.dependencies is deprecated... Use step.deps instead. 
INFO[03:48:16] Deprecated dependencies found                
INFO[03:48:16]  * [OK] Step dependency (go) installed, available. 
Quote generation failed!
Error message:
Failed to url decode response, err: invalid URL escape "% o"
exit status 1
ERRO[03:48:19] [ENVMAN] - Failed to execute command:exit status 1 
ERRO[03:48:19] Step (random-quote@3.0.0) failed, error: (exit status 1)
```

**Comment by godrei** [comment]
Hi @viktorbenei  ,

i was not able to reproduce this issue, i guess it was an issue on the service, we use to generate the quotes (`http://api.icndb.com/jokes/random`).

However the step tried to parse response body before, it checked the reponse status code. We released a new version ([3.0.2](https://github.com/bitrise-io/steps-random-quote/releases/tag/3.0.2)), which contains fix for this issue and also contains retry logic, if first request fails.

**Comment by danbtl** [comment]
@godrei The same issue just happened to me:

```
+------------------------------------------------------------------------------+
| (0) random-quote                                                             |
+------------------------------------------------------------------------------+
| id: random-quote                                                             |
| version: 3.0.2                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: bash                                                                |
| time: 2018-04-25T11:35:53-07:00                                              |
+------------------------------------------------------------------------------+
|                                                                              |
INFO[11:35:53]  * [OK] Step dependency (go) installed, available. 
Getting random quote from: http://api.icndb.com/jokes/random
Failed to url decode response (), err: invalid URL escape "% o"
exit status 1
|                                                                              |
+---+---------------------------------------------------------------+----------+
| x | random-quote (exit code: 1)                                   | 7.26 sec |
+---+---------------------------------------------------------------+----------+
| Issue tracker: https://github.com/bitrise-io/steps-random-quote/issues       |
| Source: https://github.com/bitrise-io/steps-random-quote                     |
+---+---------------------------------------------------------------+----------+
```

**Comment by viktorbenei** [comment]
@danbtl can you please create a new report? Closed issue reports are not tracked unfortunately.

**Comment by danbtl** [comment]
@viktorbenei Easy to fix, just created a pull request for this: #12

**Comment by viktorbenei** [comment]
Thanks @danbtl , really appreciate it! :)
